src/astro/sql/operators/sql_decorator.py

```diff
@@ -204,29 +208,25 @@ def handle_output_table_schema(self, output_table_name, schema=None):
         :return:
         """
         schema = schema or SCHEMA
-        if self.conn_type == "postgres" and self.schema:
+        if self.conn_type in ["postgres", "bigquery"] and self.schema:
             output_table_name = schema + "." + output_table_name
         elif self.conn_type == "snowflake" and self.schema and "." not in self.sql:
             output_table_name = self.database + "." + schema + "." + output_table_name
         return output_table_name
 
-    def _set_schema_if_needed(self):
+    def _set_schema_if_needed(self, schema=None):
         schema_statement = ""
-        if self.conn_type == "postgres":
-            schema_statement = create_schema_query(
-                conn_type=self.conn_type,
-                hook=self.hook,
-                schema_id=self.schema,
-                user=self.user,
-            )
-        elif self.conn_type == "snowflake":
+        if self.conn_type in ["postgres", "snowflake", "bigquery"]:
```

Comment:
Should we use the constants here, to avoid string comparison?

Reply:
Will address this in refactor